remotes/docker/resolver.go

```diff
@@ -231,10 +233,62 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 				return "", ocispec.Descriptor{}, errors.Wrapf(err, "%q in header not a valid digest", dgstHeader)
 			}
 			dgst = dgstHeader
+		} else {
+			log.G(ctx).Debug("no Docker-Content-Digest header, fetching manifest instead")
+
+			req, err := http.NewRequest(http.MethodGet, u, nil)
+			if err != nil {
+				return "", ocispec.Descriptor{}, err
+			}
+			req.Header = r.headers
+
+			resp, err = fetcher.doRequestWithRetries(ctx, req, nil)
+			if err != nil {
+				return "", ocispec.Descriptor{}, err
+			}
+			defer resp.Body.Close()
 		}
 
+		// Default to v1 manifest format
+		contentType := strings.SplitN(resp.Header.Get("Content-Type"), ";", 2)[0]
+		switch contentType {
+		case images.MediaTypeDockerSchema2Manifest:
+		case images.MediaTypeDockerSchema2ManifestList:
+		default:
+			contentType = images.MediaTypeDockerSchema1Manifest
+		}
+
+		// Compute the digest if the Docker-Content-Digest was not present.
 		if dgst == "" {
-			return "", ocispec.Descriptor{}, errors.Errorf("could not resolve digest for %v", ref)
+			if contentType == images.MediaTypeDockerSchema1Manifest {
+				// Schema v1 requires the signatures payload to be omitted to compute
+				// the digest.
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return "", ocispec.Descriptor{}, err
+				}
+
+				jsig, err := libtrust.ParsePrettySignature(body, "signatures")
```

Comment:
`libtrust` doesn't meet our standards for vendoring as it is deprecated and the library is not split apart to just import packages we need.

Look at the `github.com/containerd/containerd/remotes/docker/schema1` package, that has pulled out just the logic which is needed. It is ok to create a function there which just takes a schema1 blob and returns a digest for it.

Reply:
Cool thanks, added a utility method there that does what I need.